File: app/src/main/java/albion/linecutfeng/videotoascii/activity/GuideActivity.java
package albion.linecutfeng.videotoascii.activity;

import android.Manifest;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import com.luck.picture.lib.permissions.RxPermissions;

import java.io.File;

import albion.linecutfeng.videotoascii.R;
import butterknife.ButterKnife;
import butterknife.OnClick;
import io.reactivex.Observable;
import io.reactivex.ObservableEmitter;
import io.reactivex.ObservableOnSubscribe;
import io.reactivex.ObservableSource;
import io.reactivex.Observer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;
import io.reactivex.functions.Function;
import io.reactivex.schedulers.Schedulers;

import static albion.linecutfeng.videotoascii.app.AppConfig.AUDIO_PATH;
import static albion.linecutfeng.videotoascii.app.AppConfig.PIC_LIST_PATH;

public class GuideActivity extends BaseActivity {

    Handler mHander = new Handler(Looper.getMainLooper());

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.guide);
        ButterKnife.bind(this);
        checkPermissionAndMakeFile();
//        try {
//            CommonUtil.createAsciiByDrawText();
//        } catch (FileNotFoundException e) {
//            e.printStackTrace();
//        }
    }

    private void checkPermissionAndMakeFile() {
        new RxPermissions(this)
                .request(new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE})
                .subscribe(new Consumer<Boolean>() {
                    @Override
                    public void accept(Boolean granted) throws Exception {
                        if (granted) {
                            File picPath = new File(PIC_LIST_PATH);
                            File audioPath = new File(AUDIO_PATH);
                            mkUseDir(picPath);
                            mkUseDir(audioPath);
                        } else {
                            Toast.makeText(GuideActivity.this, "不给SD卡读写权限，玩毛线", Toast.LENGTH_SHORT).show();
                            mHander.postDelayed(new Runnable() {
                                @Override
                                public void run() {
                                    finish();
                                }
                            }, 1000);
                        }
                    }
                });
    }

    private void mkUseDir(File picPath) {
        if (!picPath.exists()) {
            picPath.mkdirs();
        } else if (picPath.isFile()) {
            picPath.mkdirs();
        }
    }

    @OnClick({R.id.bt_1, R.id.bt_2, R.id.bt_3, R.id.bt_4, R.id.bt_5, R.id.bt_6, R.id.bt_7, R.id.bt_8, R.id.bt_9, R.id.bt_10})
    public void onViewClicked(View view) {
        Intent intent = null;
        switch (view.getId()) {
            case R.id.bt_1:
                intent = new Intent(this, AsciiActivity.class);
                startActivity(intent);
                break;
            case R.id.bt_2:
                intent = new Intent(this, SingleProcessActivity.class);
                intent.putExtra("type", "lowpoly");
                startActivity(intent);
                break;
            case R.id.bt_3:
                intent = new Intent(this, SingleProcessActivity.class);
                intent.putExtra("type", "emoji");
                startActivity(intent);
                break;
            case R.id.bt_4:
                intent = new Intent(this, SingleProcessActivity.class);
                intent.putExtra("type", "negative");
                startActivity(intent);
                break;
            case R.id.bt_5:
                intent = new Intent(this, SingleProcessActivity.class);
                intent.putExtra("type", "casting");
                startActivity(intent);
                break;
            case R.id.bt_6:
                intent = new Intent(this, SingleProcessActivity.class);
                intent.putExtra("type", "frozen");
                startActivity(intent);
                break;

            case R.id.bt_7:
                intent = new Intent(this, SingleProcessActivity.class);
                intent.putExtra("type", "comicbook");
                startActivity(intent);
                break;

            case R.id.bt_8:
                intent = new Intent(this, SingleProcessActivity.class);
                intent.putExtra("type", "brown");
                startActivity(intent);
                break;
            case R.id.bt_9:
                intent = new Intent(this, SingleProcessActivity.class);
                intent.putExtra("type", "tilerefectrgb");
                startActivity(intent);
                break;
            case R.id.bt_10:
                intent = new Intent(this, SingleProcessActivity.class);
                intent.putExtra("type", "circleLine");
                startActivity(intent);
                break;
        }
    }

    public void wati(View view) {
        Toast.makeText(this, "敬请期待~", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/albion/linecutfeng/videotoascii/activity/GuideActivity.kt
package albion.linecutfeng.videotoascii.activity

import albion.linecutfeng.videotoascii.R
import albion.linecutfeng.videotoascii.app.AppConfig.AUDIO_PATH
import albion.linecutfeng.videotoascii.app.AppConfig.PIC_LIST_PATH
import android.Manifest
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.View
import android.widget.Toast
import com.luck.picture.lib.permissions.RxPermissions
import kotlinx.android.synthetic.main.guide.*
import java.io.File

class GuideActivity : BaseActivity(), View.OnClickListener {

    private var mHander = Handler(Looper.getMainLooper())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.guide)
        checkPermissionAndMakeFile()
        initView()
        //        try {
        //            CommonUtil.createAsciiByDrawText();
        //        } catch (FileNotFoundException e) {
        //            e.printStackTrace();
        //        }
    }

    private fun initView() {
        bt_1.setOnClickListener(this)
        bt_2.setOnClickListener(this)
        bt_3.setOnClickListener(this)
        bt_4.setOnClickListener(this)
        bt_5.setOnClickListener(this)
        bt_6.setOnClickListener(this)
        bt_7.setOnClickListener(this)
        bt_8.setOnClickListener(this)
        bt_9.setOnClickListener(this)
        bt_10.setOnClickListener(this)
        bt_11.setOnClickListener(this)
    }

    private fun checkPermissionAndMakeFile() {
        var subscribe = RxPermissions(this)
                .request(Manifest.permission.WRITE_EXTERNAL_STORAGE)
                .subscribe { granted ->
                    if (granted == true) {
                        val picPath = File(PIC_LIST_PATH)
                        val audioPath = File(AUDIO_PATH)
                        mkUseDir(picPath)
                        mkUseDir(audioPath)
                    } else {
                        Toast.makeText(this@GuideActivity, "不给SD卡读写权限，玩毛线", Toast.LENGTH_SHORT).show()
                        mHander.postDelayed({ finish() }, 1000)
                    }
                }
    }

    private fun mkUseDir(picPath: File) {
        if (!picPath.exists()) {
            picPath.mkdirs()
        } else if (picPath.isFile) {
            picPath.mkdirs()
        }
    }

    override fun onClick(view: View?) {
        val intent: Intent?
        when (view?.id) {
            R.id.bt_1 -> {
                intent = Intent(this, AsciiActivity::class.java)
                startActivity(intent)
            }
            R.id.bt_2 -> {
                intent = Intent(this, SingleProcessActivity::class.java)
                intent.putExtra("type", "lowpoly")
                startActivity(intent)
            }
            R.id.bt_3 -> {
                intent = Intent(this, SingleProcessActivity::class.java)
                intent.putExtra("type", "emoji")
                startActivity(intent)
            }
            R.id.bt_4 -> {
                intent = Intent(this, SingleProcessActivity::class.java)
                intent.putExtra("type", "negative")
                startActivity(intent)
            }
            R.id.bt_5 -> {
                intent = Intent(this, SingleProcessActivity::class.java)
                intent.putExtra("type", "casting")
                startActivity(intent)
            }
            R.id.bt_6 -> {
                intent = Intent(this, SingleProcessActivity::class.java)
                intent.putExtra("type", "frozen")
                startActivity(intent)
            }

            R.id.bt_7 -> {
                intent = Intent(this, SingleProcessActivity::class.java)
                intent.putExtra("type", "comicbook")
                startActivity(intent)
            }

            R.id.bt_8 -> {
                intent = Intent(this, SingleProcessActivity::class.java)
                intent.putExtra("type", "brown")
                startActivity(intent)
            }
            R.id.bt_9 -> {
                intent = Intent(this, SingleProcessActivity::class.java)
                intent.putExtra("type", "tilerefectrgb")
                startActivity(intent)
            }
            R.id.bt_10 -> {
                intent = Intent(this, SingleProcessActivity::class.java)
                intent.putExtra("type", "circleLine")
                startActivity(intent)
            }
            R.id.bt_11 -> {
                intent = Intent(this,  DFTProcessActivity::class.java)
                startActivity(intent)
            }
        }
    }

    fun wati(view: View?) {
        Toast.makeText(this, "敬请期待~", Toast.LENGTH_SHORT).show()
    }
}
